spring-vault-core/src/main/java/org/springframework/vault/support/EcPrivateKeyStrategy.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.vault.support;
+
+import java.security.spec.ECFieldFp;
+import java.security.spec.ECParameterSpec;
+import java.security.spec.ECPoint;
+import java.security.spec.ECPrivateKeySpec;
+import java.security.spec.EllipticCurve;
+import java.security.spec.KeySpec;
+import org.bouncycastle.asn1.ASN1ObjectIdentifier;
```

Comment:
We use Bouncy Castle here for EC support. Can we use this dependency here?

Reply:
Ideally, we don't use BouncyCastle at all as the cost for the dependency is quite large in comparison to what we get. We have a DER parser already and it's a matter of parsing the ASN.1 sequence. Java's crypto utils contain proper support to resolve EC parameters through an OID and provide key specs.

I have the code ready for this.